### PR TITLE
Fix migrate (regression from #2520)

### PR DESCRIPTION
**Pull request by ykzts**
#2520 is broken migration.

<details>
<summary><code>$ docker-compose run --rm web ./bin/rails db:migrate</code></summary>
<pre><code>Migrating to CreateBlockMutes (20180428000000)
== 20180428000000 CreateBlockMutes: migrating =================================
-- create_table("block_mutes", {:force=>:casecade})
   -> 0.0187s
-- add_column(:blocks, :block, :boolean, {:null=>false})
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::NotNullViolation: ERROR:  column "block" contains null values
: ALTER TABLE "blocks" ADD "block" boolean NOT NULL
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:98:in `async_exec'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:98:in `block in execute'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:589:in `block in log'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:583:in `log'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:97:in `execute'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/schema_statements.rb:542:in `add_column'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/schema_statements.rb:477:in `add_column'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:846:in `block in method_missing'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:815:in `block in say_with_time'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:815:in `say_with_time'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:835:in `method_missing'
/mastodon/db/migrate/20180428000000_create_block_mutes.rb:11:in `change'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:789:in `exec_migration'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:773:in `block (2 levels) in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:772:in `block in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:398:in `with_connection'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:771:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:951:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1214:in `block in execute_migration_in_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1282:in `block in ddl_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `block in transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/transaction.rb:189:in `within_new_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/transactions.rb:211:in `transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1282:in `ddl_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1213:in `execute_migration_in_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1185:in `block in migrate_without_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1184:in `each'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1184:in `migrate_without_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1132:in `block in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1301:in `with_advisory_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1132:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1006:in `up'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:984:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/tasks/database_tasks.rb:161:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/railties/databases.rake:58:in `block (2 levels) in <top (required)>'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/rake_proxy.rb:14:in `block in run_rake_task'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/rake_proxy.rb:11:in `run_rake_task'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/commands_tasks.rb:51:in `run_command!'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands.rb:18:in `<top (required)>'
./bin/rails:4:in `require'
./bin/rails:4:in `<main>'
ActiveRecord::StatementInvalid: PG::NotNullViolation: ERROR:  column "block" contains null values
: ALTER TABLE "blocks" ADD "block" boolean NOT NULL
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:98:in `async_exec'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:98:in `block in execute'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:589:in `block in log'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:583:in `log'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:97:in `execute'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/schema_statements.rb:542:in `add_column'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/schema_statements.rb:477:in `add_column'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:846:in `block in method_missing'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:815:in `block in say_with_time'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:815:in `say_with_time'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:835:in `method_missing'
/mastodon/db/migrate/20180428000000_create_block_mutes.rb:11:in `change'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:789:in `exec_migration'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:773:in `block (2 levels) in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:772:in `block in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:398:in `with_connection'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:771:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:951:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1214:in `block in execute_migration_in_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1282:in `block in ddl_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `block in transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/transaction.rb:189:in `within_new_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/transactions.rb:211:in `transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1282:in `ddl_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1213:in `execute_migration_in_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1185:in `block in migrate_without_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1184:in `each'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1184:in `migrate_without_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1132:in `block in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1301:in `with_advisory_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1132:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1006:in `up'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:984:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/tasks/database_tasks.rb:161:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/railties/databases.rake:58:in `block (2 levels) in <top (required)>'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/rake_proxy.rb:14:in `block in run_rake_task'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/rake_proxy.rb:11:in `run_rake_task'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/commands_tasks.rb:51:in `run_command!'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands.rb:18:in `<top (required)>'
./bin/rails:4:in `require'
./bin/rails:4:in `<main>'
PG::NotNullViolation: ERROR:  column "block" contains null values
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:98:in `async_exec'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:98:in `block in execute'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:589:in `block in log'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activesupport-5.0.2/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract_adapter.rb:583:in `log'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/database_statements.rb:97:in `execute'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/schema_statements.rb:542:in `add_column'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/postgresql/schema_statements.rb:477:in `add_column'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:846:in `block in method_missing'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:815:in `block in say_with_time'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:815:in `say_with_time'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:835:in `method_missing'
/mastodon/db/migrate/20180428000000_create_block_mutes.rb:11:in `change'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:789:in `exec_migration'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:773:in `block (2 levels) in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:772:in `block in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/connection_pool.rb:398:in `with_connection'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:771:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:951:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1214:in `block in execute_migration_in_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1282:in `block in ddl_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `block in transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/transaction.rb:189:in `within_new_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/connection_adapters/abstract/database_statements.rb:232:in `transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/transactions.rb:211:in `transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1282:in `ddl_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1213:in `execute_migration_in_transaction'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1185:in `block in migrate_without_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1184:in `each'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1184:in `migrate_without_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1132:in `block in migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1301:in `with_advisory_lock'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1132:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:1006:in `up'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/migration.rb:984:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/tasks/database_tasks.rb:161:in `migrate'
/mastodon/vendor/bundle/ruby/2.4.0/gems/activerecord-5.0.2/lib/active_record/railties/databases.rake:58:in `block (2 levels) in <top (required)>'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/rake_proxy.rb:14:in `block in run_rake_task'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/rake_proxy.rb:11:in `run_rake_task'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands/commands_tasks.rb:51:in `run_command!'
/mastodon/vendor/bundle/ruby/2.4.0/gems/railties-5.0.2/lib/rails/commands.rb:18:in `<top (required)>'
./bin/rails:4:in `require'
./bin/rails:4:in `<main>'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)</code></pre>
</details>

### CODE REVIEW

**Comment by ykzts** [comment]
@akihikodaki please check.

File: db/migrate/20170428000001_change_blocks_and_mutes.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class ChangeBlocksAndMutes < ActiveRecord::Migration[5.0]
+  def up
+    add_column       :blocks, :block, :boolean
+    change_column    :blocks, :block, :boolean, null: false
```
Comment (akihikodaki) [inline db/migrate/20170428000001_change_blocks_and_mutes.rb]: Why not `add_column :blocks, :block, :boolean, null: false`?

File: db/schema.rb [hunk 2]
```diff
@@ -70,7 +70,6 @@
     t.datetime "updated_at",        null: false
     t.index ["account_id", "target_account_id"], name: "index_blocks_on_account_id_and_target_account_id", unique: true, using: :btree
   end
-  execute   "ALTER TABLE blocks ADD CONSTRAINT check_blocks_on_block CHECK(block = TRUE), INHERIT block_mutes"
```
Comment (akihikodaki) [inline db/schema.rb]: Please leave this line as it is.

File: db/schema.rb [hunk 3]
```diff
@@ -54,7 +54,7 @@
     t.index ["username", "domain"], name: "index_accounts_on_username_and_domain", unique: true, using: :btree
   end
 
-  create_table "block_mutes", force: :cascade, id: false do |t|
+  create_table "block_mutes", force: :cascade do |t|
```
Comment (akihikodaki) [inline db/schema.rb]: It is not good to have `id` column in `block_mutes`.
> This table will contain no data. Do not define any check constraints on this table, unless you intend them to be applied equally to all partitions. There is no point in defining any indexes or unique constraints on it, either.

[PostgreSQL: Documentation: 9.6: Partitioning](https://www.postgresql.org/docs/9.6/static/ddl-partitioning.html)

**Review by akihikodaki — CHANGES_REQUESTED** [review]
Sorry for bothering with my mistake, and thank you for the fix. I would like to ask some questions about your change in this review.

File: db/schema.rb (hunk 2 above)
Comment (ykzts) [inline db/schema.rb]: This SQL statement is required only during migration. I think Schema does not need it, how is it?

Comment (akihikodaki) [inline db/schema.rb]: It needs. `CHECK` constraint and inheritance could not be expressed with ActionCable (without any extension).

File: db/schema.rb (hunk 3 above)
Comment (ykzts) [inline db/schema.rb]: If so, migration may also be necessary. Because it was not there, I left it as it was.

Comment (akihikodaki) [inline db/schema.rb]: That is my fault. The migrator is broken.

File: db/migrate/20170428000001_change_blocks_and_mutes.rb (hunk 1 above)
Comment (akihikodaki) [inline db/migrate/20170428000001_change_blocks_and_mutes.rb]: I guess change_column is intended to be after Block.update_all?

**Comment by unarist** [comment]
It still fails if block record exists before migration:

```
PG::NotNullViolation: ERROR:  column "block" contains null values
: ALTER TABLE "blocks" ALTER "block" SET NOT NULL
```

```
postgres=# select * from blocks;
 id | account_id | target_account_id |         created_at         |         updated_at
----+------------+-------------------+----------------------------+----------------------------
  1 |          1 |                 4 | 2017-04-30 05:04:10.963466 | 2017-04-30 05:04:10.963466
```

When you added `block` column as nullable, its value will be `NULL`. So you must set actual values before making the column non-nullable.

**Review by akihikodaki — APPROVED** [review]

**Review by akihikodaki — CHANGES_REQUESTED** [review]
Could it be suqashd?

**Comment by ykzts** [comment]
I will do it when all is over.

File: db/migrate/20170428000001_change_blocks_and_mutes.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+class ChangeBlocksAndMutes < ActiveRecord::Migration[5.0]
+  def up
+    add_column       :blocks, :block, :boolean
+    execute          "ALTER TABLE blocks ADD CONSTRAINT check_mutes_on_block CHECK(block = TRUE), INHERIT block_mutes"
+    Block.update_all block: true
+    change_column    :blocks, :block, :boolean, null: false
```
Comment (unarist) [inline db/migrate/20170428000001_change_blocks_and_mutes.rb]:
```
PG::DatatypeMismatch: ERROR:  column "block" in child table must be marked NOT NULL
: ALTER TABLE blocks ADD CONSTRAINT check_mutes_on_block CHECK(block = TRUE), INHERIT block_mutes
```

Probably, you should do with below order:

1. add_column
2. Block.update_all
3. change_column
4. execute

**Review by unarist — CHANGES_REQUESTED** [review]
Please do test with non-empty `blocks` table...

File: db/migrate/20170428000001_change_blocks_and_mutes.rb (hunk 4 above)
Comment (akihikodaki) [inline db/migrate/20170428000001_change_blocks_and_mutes.rb]: Exactly. I will also test on my instance when this work has finished.

File: db/schema.rb (hunk 2 above)
Comment (unarist) [inline db/schema.rb]: @akihikodaki Well, did you added this line manually in your PR? If so, it will be removed all future db changes...

Comment (akihikodaki) [inline db/schema.rb]: Yes, that will kill the auto generation of the database.

**Comment by akihikodaki** [comment]
Could you remove the comment at the top of `db/schema.rb`? This file could not be auto generated since we do not have PostgreSQL extensions for ActiveRecord.

**Comment by ykzts** [comment]
@akihikodaki That comment is automatically generated. Therefore, when someone next performs db: migrate, it is added again.

**Comment by akihikodaki** [comment]
Well, really? I need to study ActiveRecord more.

**Comment by akihikodaki** [comment]
Opened #2634 to revert #2520.

**Comment by ykzts** [comment]
Yes. I think that I should do so too.

**Comment by akihikodaki** [comment]
I wonder the problem could be fixed by altering `config.active_record.schema_format` to `:sql`.

**Pull request closed without merging**